src/Symfony/Bridge/Doctrine/Validator/Constraints/UniqueEntityValidator.php

```diff
@@ -64,32 +64,37 @@ public function validate($entity, Constraint $constraint)
             return;
         }
 
+        $entityClass = $constraint->entityClass ?: get_class($entity);
+
         if ($constraint->em) {
             $em = $this->registry->getManager($constraint->em);
 
             if (!$em) {
                 throw new ConstraintDefinitionException(sprintf('Object manager "%s" does not exist.', $constraint->em));
             }
         } else {
-            $em = $this->registry->getManagerForClass(get_class($entity));
+            $em = $this->registry->getManagerForClass($entityClass);
 
             if (!$em) {
                 throw new ConstraintDefinitionException(sprintf('Unable to find the object manager associated with an entity of class "%s".', get_class($entity)));
             }
         }
 
-        $class = $em->getClassMetadata(get_class($entity));
-        /* @var $class \Doctrine\Common\Persistence\Mapping\ClassMetadata */
+        $class = $em->getClassMetadata($entityClass);
 
         $criteria = array();
         $hasNullValue = false;
 
-        foreach ($fields as $fieldName) {
-            if (!$class->hasField($fieldName) && !$class->hasAssociation($fieldName)) {
-                throw new ConstraintDefinitionException(sprintf('The field "%s" is not mapped by Doctrine, so it cannot be validated for uniqueness.', $fieldName));
+        foreach ($fields as $formField => $entityField) {
+            if (!$class->hasField($entityField) && !$class->hasAssociation($entityField)) {
+                throw new ConstraintDefinitionException(sprintf('The field "%s" is not mapped by Doctrine, so it cannot be validated for uniqueness.', $entityField));
             }
 
-            $fieldValue = $class->reflFields[$fieldName]->getValue($entity);
+            $field = new \ReflectionProperty($entityClass, is_int($formField) ? $entityField : $formField);
```

Comment:
This seems wrong, `$entityClass` is not the class of the `$entity` subject always, should be `get_class($entity)`, no?

Reply:
Yes indeed. Following [your suggestion](https://github.com/symfony/symfony/pull/24974#discussion_r151177388) I'll rename `$entity` `$object` this will be clearer